### PR TITLE
Use `stringer` for "enums"

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Makes debugging easier.

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/types/path.go [hunk 1]
```diff
@@ -27,20 +27,29 @@ import (
...
+//go:generate ../../bin/stringer -linecomment -type DocumentPathErrorCode
+
 // DocumentPathErrorCode represents DocumentPathError error code.
 type DocumentPathErrorCode int
 
 const (
+	_ DocumentPathErrorCode = iota
+
 	// ErrDocumentPathKeyNotFound indicates that key was not found in document.
-	ErrDocumentPathKeyNotFound = iota + 1
```
Comment (AlekSi) [inline internal/types/path.go:35]: Those constants were untyped numbers, not `DocumentPathErrorCode`

**Review by rumyantseva — APPROVED** [review]
Looks reasonable to me (the linter complains a bit as it wants all the error codes mentioned explicitly in the switch).

**Review by noisersup — DISMISSED** [review]
LGTM

**Review by chilagrow — DISMISSED** [review]

**Review by w84thesun — APPROVED** [review]
Looks good, although the linter complains about that switch statement.

**Review by chilagrow — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**